### DISCUSSION TITLE
z.keyof()?

### DISCUSSION THREAD

**Discussion by ftzi** [colinhacks/zod#2393] [Ideas]
I have an object. I want to validate that the given string is a keyof that object I have.

I tried the following:

`z.enum(Object.keys(Items))`
`z.enum(Object.keys(Items) as (keyof typeof Items)[])`

but they won't work. There is this answer elsewhere but it won't work: https://github.com/colinhacks/zod/discussions/839#discussioncomment-4335236

1) How can I achieve it in the best way?
2) There is the `z.object({}).keyof()`. Should Zod have the `z.keyof()` to achieve what I am trying to do? Seems a good thing to have in the library.

Edit: I managed to get this working `z.custom<keyof typeof Items>((v) => Object.keys(Items).includes(v as string))`, but I still think we should have `z.keyof`!

**Comment by 0xethen** [comment]
for anyone in the future, I believe this is possible with `z.keyof(z.object(Items))` in Zod Mini v4 (`Items` being the object)
